Framework/DataHandling/src/LoadMuonNexusV2.cpp

```diff
@@ -0,0 +1,227 @@
+    return 0;
+  ::NeXus::File file(descriptor.getFilename());
+  file.openPath(root + "/definition");
+  std::string def = file.getStrData();
+  if (def == "muonTD" || def == "pulsedTD") {
+    return 82; // have to return 82 to "beat" the LoadMuonNexus2 algorithm,
+               // which returns 81 for this file as well
+  } else {
+    return 0;
+  }
+}
+/// Initialization method.
+void LoadMuonNexusV2::init() {
+
+  std::vector<std::string> extensions{".nxs", ".nxs_v2"};
+  declareProperty(std::make_unique<FileProperty>(
+                      "Filename", "", FileProperty::Load, extensions),
+                  "The name of the Nexus file to load");
+
+  declareProperty(
+      std::make_unique<WorkspaceProperty<Workspace>>("OutputWorkspace", "",
+                                                     Direction::Output),
+      "The name of the workspace to be created as the output of the\n"
+      "algorithm. For multiperiod files, one workspace will be\n"
+      "generated for each period");
+
+  auto mustBePositive = std::make_shared<BoundedValidator<int64_t>>();
+  mustBePositive->setLower(0);
+  declareProperty("SpectrumMin", static_cast<int64_t>(0), mustBePositive);
+  declareProperty("SpectrumMax", static_cast<int64_t>(EMPTY_INT()),
+                  mustBePositive);
+  declareProperty(std::make_unique<ArrayProperty<int64_t>>("SpectrumList"));
+  declareProperty("EntryNumber", static_cast<int64_t>(0), mustBePositive,
+                  "0 indicates that every entry is loaded, into a separate "
+                  "workspace within a group. "
+                  "A positive number identifies one entry to be loaded, into "
+                  "one workspace");
+
+  std::vector<std::string> FieldOptions{"Transverse", "Longitudinal"};
+  declareProperty("MainFieldDirection", "Transverse",
+                  std::make_shared<StringListValidator>(FieldOptions),
+                  "Output the main field direction if specified in Nexus file "
+                  "(default longitudinal).",
+                  Direction::Output);
+
+  declareProperty("TimeZero", 0.0,
+                  "Time zero in units of micro-seconds (default to 0.0)",
+                  Direction::Output);
+  declareProperty("FirstGoodData", 0.0,
+                  "First good data in units of micro-seconds (default to 0.0)",
+                  Direction::Output);
+
+  declareProperty(
+      std::make_unique<WorkspaceProperty<Workspace>>(
+          "DeadTimeTable", "", Direction::Output, PropertyMode::Optional),
+      "Table or a group of tables containing detector dead times.");
+
+  declareProperty(std::make_unique<WorkspaceProperty<Workspace>>(
+                      "DetectorGroupingTable", "", Direction::Output,
+                      PropertyMode::Optional),
+                  "Table or a group of tables with information about the "
+                  "detector grouping.");
+}
+void LoadMuonNexusV2::execLoader() {
+  this->setRethrows(true);
```

Comment:
It's unusual to set this inside an algorithm as only the caller generally knows how the algorithm is used and whether it is safe to rethrow an exception. Can you help me understand this requirement?

Reply:
No good reason for this to rethrow, for some reason I had made a test which relied on it rethrowing from LoadISISNexus, which was really uncessary.

Reply:
Ok. `createChildAlgorithm` sets rethrows as appropriate so that should all be okay.